### PR TITLE
Add a system_auditor role who has read on everything.

### PR DESCRIPTION
**Pull request by jctanner**
https://issues.redhat.com/browse/AAP-24730

```
jtanner@p1:~/workspace/github/jctanner.redhat/galaxy_ng.system_auditor$ curl -L -u bob:redhat1234 http://localhost:55001/api/galaxy/_ui/v1/users/ | jq .                                                                                                                                                                      
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   117  100   117    0     0    338      0 --:--:-- --:--:-- --:--:--   339
{
  "errors": [
    {
      "status": "403",
      "code": "permission_denied",
      "title": "You do not have permission to perform this action."
    }
  ]
}
jtanner@p1:~/workspace/github/jctanner.redhat/galaxy_ng.system_auditor$ curl -L -u sa:redhat1234 http://localhost:55001/api/galaxy/_ui/v1/users/ | jq .                                                                                                                                                                      
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   780  100   780    0     0   2090      0 --:--:-- --:--:-- --:--:--  2091
{
  "meta": {
    "count": 3
  },
  "links": {
    "first": "/api/galaxy/_ui/v1/users/?limit=10&offset=0",
    "previous": null,
    "next": null,
    "last": "/api/galaxy/_ui/v1/users/?limit=10&offset=0"
  },
  "data": [
    {
      "id": 1,
      "username": "admin",
      "first_name": "",
      "last_name": "",
      "email": "admin@example.com",
      "groups": [],
      "date_joined": "2024-06-06T19:46:09.181235Z",
      "is_superuser": true,
      "auth_provider": [
        "django"
      ]
    },
    {
      "id": 2,
      "username": "sa",
      "first_name": "system",
      "last_name": "auditor",
      "email": "sa@localhost",
      "groups": [
        {
          "id": 1,
          "name": "sa-group"
        }
      ],
      "date_joined": "2024-06-06T19:59:29.183763Z",
      "is_superuser": false,
      "auth_provider": [
        "django"
      ]
    },
    {
      "id": 3,
      "username": "bob",
      "first_name": "bob",
      "last_name": "barker",
      "email": "bob@localhost",
      "groups": [],
      "date_joined": "2024-06-06T20:12:02.936366Z",
      "is_superuser": false,
      "auth_provider": [
        "django"
      ]
    }
  ]
}
jtanner@p1:~/workspace/github/jctanner.redhat/galaxy_ng.system_auditor$ curl -X POST -H 'Content-Type:application/json' -d '{"username": "foo"}' -u sa:redhat1234 http://localhost:55001/api/galaxy/_ui/v1/users/ | jq .                                                                                                      
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   136  100   117  100    19    341     55 --:--:-- --:--:-- --:--:--   396
{
  "errors": [
    {
      "status": "403",
      "code": "permission_denied",
      "title": "You do not have permission to perform this action."
    }
  ]
}
```

### CODE REVIEW

**Review by chr-stian — APPROVED** [review]

File: galaxy_ng/app/migrations/0053_create_system_auditor_role.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+from django.db import migrations
+
+
+def create_system_auditor_role(apps, schema_editor):
+    Role = apps.get_model("core", "Role")
+    Permission = apps.get_model("auth", "Permission")
+
+    # Create the role
+    role, created = Role.objects.get_or_create(
+        name='galaxy.system_auditor',
```
Comment (AlanCoding) [inline galaxy_ng/app/migrations/0053_create_system_auditor_role.py:12]: Philosophically, reading into the intent behind the naming here, `galaxy.*` suggests to me this is _not_ intended to be a shared role. Practically, we need an auditor role to be a shared role. For models and permissions, we name things with `shared.*`. So maybe you could name it `shared.system_auditor` or `shared.cluster_auditor` or something like that. That would suggest that naming of other roles would be like `shared.organization_admin`, which I have not looked into yet.

If you're introducing a auditor role that is _not_ shared, then I don't have anything to add, other than that's not a requirement coming from my side.

**Review by bmclaughlin — APPROVED** [review]

File: galaxy_ng/app/migrations/0053_create_system_auditor_role.py (hunk 1 above)
Comment (newswangerd) [reply galaxy_ng/app/migrations/0053_create_system_auditor_role.py:12]: @AlanCoding this is a pulp thing. We have to prefix our roles with `galaxy.`.

**Review by AlanCoding — APPROVED** [review]

**Review by newswangerd — CHANGES_REQUESTED** [review]
All of our default role definition need to go in this file: https://github.com/ansible/galaxy_ng/blob/master/galaxy_ng/app/access_control/statements/roles.py

Pulp has a post migration script that picks up that file and creates or updates the default roles with it.

**Comment by jctanner** [comment]
replaced by https://github.com/ansible/galaxy_ng/pull/2162

**Pull request closed without merging**